tests/testthat/test-github-action.R

```diff
@@ -0,0 +1,43 @@
+test_that("use_github_action() allows for custom urls", {
+  skip_if_no_git_user()
+
+  scoped_temporary_package()
+  use_git()
+  use_git_remote(name = "origin", url = "https://github.com/fake/fake")
+
+  # Directly call to r-lib actions
+  use_github_action(url = "https://raw.githubusercontent.com/r-lib/actions/master/examples/check-full.yaml")
+  expect_proj_dir(".github")
+  expect_proj_dir(".github/workflows")
+  expect_proj_file(".github/workflows/check-full.yaml")
+
+})
+
+test_that("use_github_action() replaces yml with yaml in name", {
```

Comment:
If I'm right above, I think this can be added as a second `use_github_action()` call in the test above, with `.yml` instead of `.yaml`.

Reply:
The `.yml` / `.yaml` components would only be enforced when there isn't a custom URL. So, I don't think this can be merged in with the above test as it is checking:

```r
use_github_action("pkgdown.yml")
```

whereas the first test is ensuring that a custom URL can be pulled and downloaded.

```r
use_github_action(url="https://my-secret/gh/actions/master/examples/pkgdown.yaml")
```